Interop/ONNX/src/test/java/org/tribuo/interop/onnx/extractors/BERTFeatureExtractorTest.java

```diff
@@ -0,0 +1,136 @@
+import org.tribuo.classification.LabelFactory;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+
+public class BERTFeatureExtractorTest {
+
+    @Test
+    public void testTokenizerLoading() throws URISyntaxException, IOException {
+        Path vocabPath = Paths.get(BERTFeatureExtractorTest.class.getResource("bert-base-cased-vocab.txt").toURI());
+        Path tokenizerPath = Paths.get(BERTFeatureExtractorTest.class.getResource("bert-base-cased-tokenizer.json").toURI());
+        BERTFeatureExtractor.TokenizerConfig config = BERTFeatureExtractor.loadTokenizer(tokenizerPath);
+
+        List<String> vocabList = Files.readAllLines(vocabPath, StandardCharsets.UTF_8);
+
+        Assertions.assertEquals(config.tokenIDs.size(),vocabList.size());
+
+        for (String vocabElement : vocabList) {
+            Assertions.assertTrue(config.tokenIDs.containsKey(vocabElement));
+        }
+
+        Assertions.assertEquals(100, config.maxInputCharsPerWord);
+        Assertions.assertEquals(false, config.lowercase);
+        Assertions.assertEquals(false, config.stripAccents);
+        Assertions.assertEquals("[UNK]",config.unknownToken);
+        Assertions.assertEquals("[CLS]",config.classificationToken);
+        Assertions.assertEquals("[SEP]",config.separatorToken);
+
+        tokenizerPath = Paths.get(BERTFeatureExtractorTest.class.getResource("tinybert-tokenizer.json").toURI());
+        config = BERTFeatureExtractor.loadTokenizer(tokenizerPath);
+        Assertions.assertEquals(100, config.maxInputCharsPerWord);
+        Assertions.assertEquals(true, config.lowercase);
+        Assertions.assertEquals(true, config.stripAccents);
+        Assertions.assertEquals("[UNK]",config.unknownToken);
+        Assertions.assertEquals("[CLS]",config.classificationToken);
+        Assertions.assertEquals("[SEP]",config.separatorToken);
+    }
+
+    @Test
+    public void testEmbedding() throws URISyntaxException, OrtException {
+        // tinybert's vocab is build on the first seven chapters of Pride and Prejudice from project gutenberg
+        // and has 1656 elements.
+        Path tokenizerPath = Paths.get(BERTFeatureExtractorTest.class.getResource("tinybert-tokenizer.json").toURI());
+        /* tinybert is untrained, built with the following config
+        config = BertConfig(
+                vocab_size=tokenizer.vocab_size,
+                hidden_size=50,
+                num_hidden_layers=1,
+                num_attention_heads=1,
+                intermediate_size=200,
+                max_position_embeddings=100
+        )
+        */
+        Path modelPath = Paths.get(BERTFeatureExtractorTest.class.getResource("tinybert.onnx").toURI());
+        LabelFactory factory = new LabelFactory();
+        String input = "It is a truth universally acknowledged,";
```

Comment:
I'd maybe add a couple more tests to check out-of-vocab behavior (e.g. Arabic or CJK characters, emoji) just so we know what happens. I'm not sure what's OOV for the full bert-base-cased tokenizer but should be easy to check with tinybert.

Reply:
There are tests for out-of-vocab wordpiece tokenizer behaviour in the tokens module.